interoperator/controllers/schedulers/sflabelselectorscheduler/sflabelselectorscheduler_controller.go

```diff
@@ -0,0 +1,245 @@
+			// Object not found, return.  Created objects are automatically garbage collected.
+			// For additional cleanup logic use finalizers.
+			return ctrl.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return ctrl.Result{}, err
+	}
+
+	if instance.Spec.ClusterID == "" {
+		labelSelector, err := getLabelSelectorString(*instance, *r)
+		if err != nil || labelSelector == "" {
+			log.Info("Failed to get labelSelector string..", "error", err, "labelSelector", labelSelector)
+			return ctrl.Result{}, err
+		}
+
+		clusterID, err := r.schedule(labelSelector)
+		if err != nil {
+			log.Error(err, "Failed to schedule ", "labelSelector", labelSelector, "clusterID", clusterID)
+			return ctrl.Result{}, err
+		}
+
+		if clusterID != "" {
+			log.Info("setting clusterID", "clusterID", clusterID)
+			instance.Spec.ClusterID = clusterID
+			if err := r.Update(ctx, instance); err != nil {
+				log.Error(err, "failed to set cluster id", "clusterID", clusterID)
+				return ctrl.Result{}, err
+			}
+		}
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func getLabelSelectorString(sfServiceInstance osbv1alpha1.SFServiceInstance, r SFLabelSelectorScheduler) (string, error) {
+	log := r.Log
+	ctx := context.Background()
+
+	sfNamespace := os.Getenv(constants.NamespaceEnvKey)
+	if sfNamespace == "" {
+		sfNamespace = constants.DefaultServiceFabrikNamespace
+	}
+	plan := &osbv1alpha1.SFPlan{}
+	namespacedName := types.NamespacedName{
+		Name:      sfServiceInstance.Spec.PlanID,
+		Namespace: sfNamespace,
+	}
+	err := r.Get(ctx, namespacedName, plan)
+	if err != nil {
+		return "", err
+	}
+	service := &osbv1alpha1.SFService{}
+	namespacedName = types.NamespacedName{
+		Name:      sfServiceInstance.Spec.ServiceID,
+		Namespace: sfNamespace,
+	}
+	err = r.Get(ctx, namespacedName, service)
+	if err != nil {
+		return "", err
+	}
+
+	labelSelectorTemplate, err := plan.GetTemplate(osbv1alpha1.ClusterLabelSelectorAction)
+	if err != nil || labelSelectorTemplate.Type != "gotemplate" {
+		log.Info("plan does not have clusterlabel template")
+		return "", err
+	}
+
+	content := labelSelectorTemplate.Content
+	values := make(map[string]interface{})
+	instanceObj, err := dynamic.ObjectToMapInterface(sfServiceInstance)
+	values["instance"] = instanceObj
+	if err != nil {
+		return "", err
+	}
+
+	if service != nil {
+		serviceObj, err := dynamic.ObjectToMapInterface(service)
+		values["service"] = serviceObj
+		if err != nil {
+			return "", err
+		}
+	}
+	if plan != nil {
+		planObj, err := dynamic.ObjectToMapInterface(plan)
+		values["plan"] = planObj
+		if err != nil {
+			return "", err
+		}
+	}
+
+	renderer, err := gotemplate.New()
+	if err != nil {
+		return "", err
+	}
+	rendererInput := gotemplate.NewInput("", content, "test", values)
+	rendererOutput, err := renderer.Render(rendererInput)
+	if err != nil {
+		return "", err
+	}
+	labelSelector, err := rendererOutput.FileContent("main")
+	if err != nil {
+		return "", err
+	}
```

Comment:
We can use `rendererFactory` here. 
`rendererFactory.GetRenderer(labelSelectorTemplate.Type, nil)`
`rendererFactory.GetRendererInput`
and
`renderer.Render`